lock-api/src/test/java/com/palantir/lock/client/MultiClientCommitTimestampGetterTest.java

```diff
@@ -0,0 +1,251 @@
+
+    private final Consumer<List<BatchElement<NamespacedRequest, Long>>> consumer =
+            MultiClientCommitTimestampGetter.consumer(timelockService);
+
+    @Test
+    public void canServiceOneClient() {
+        setupServiceAndAssertSanityOfResponse(getCommitTimestampRequestsForClients(1, COMMIT_TS_LIMIT_PER_REQUEST - 1));
+    }
+
+    @Test
+    public void canServiceOneClientInMultipleRequests() {
+        setupServiceAndAssertSanityOfResponse(getCommitTimestampRequestsForClients(1, COMMIT_TS_LIMIT_PER_REQUEST * 5));
+    }
+
+    @Test
+    public void canServiceMultipleClients() {
+        int clientCount = 50;
+        setupServiceAndAssertSanityOfResponse(
+                getCommitTimestampRequestsForClients(clientCount, (COMMIT_TS_LIMIT_PER_REQUEST - 1) * clientCount));
+    }
+
+    @Test
+    public void canServiceMultipleClientsWithMultipleServerCalls() {
+        int clientCount = 5;
+        setupServiceAndAssertSanityOfResponse(
+                getCommitTimestampRequestsForClients(clientCount, (COMMIT_TS_LIMIT_PER_REQUEST + 1) * clientCount));
+    }
+
+    @Test
+    public void updatesCacheWhileProcessingResponse() {
+        Namespace client = Namespace.of("Kitty");
+        List<BatchElement<NamespacedRequest, Long>> batchElements = IntStream.range(0, COMMIT_TS_LIMIT_PER_REQUEST * 2)
+                .mapToObj(ind -> batchElementForNamespace(client))
+                .collect(toList());
+        setupServiceAndAssertSanityOfResponse(batchElements);
+
+        LockWatchEventCache cache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(client);
+        verify(cache, times(2)).lastKnownVersion();
+        verify(cache, times(2)).processGetCommitTimestampsUpdate(any(), any());
+    }
+
+    @Test
+    public void doesNotUpdateCacheIfClientNotServed() {
+        Namespace alpha = Namespace.of("alpha" + UUID.randomUUID());
+        Namespace beta = Namespace.of("beta" + UUID.randomUUID());
+
+        BatchElement<NamespacedRequest, Long> requestForAlpha = batchElementForNamespace(alpha);
+        BatchElement<NamespacedRequest, Long> requestForBeta = batchElementForNamespace(beta);
+
+        List<BatchElement<NamespacedRequest, Long>> allRequests = ImmutableList.of(requestForAlpha, requestForBeta);
+        List<BatchElement<NamespacedRequest, Long>> alphaRequestList = ImmutableList.of(requestForAlpha);
+        Map<Namespace, GetCommitTimestampsResponse> responseMap = getCommitTimestamps(alphaRequestList);
+
+        when(timelockService.getCommitTimestamps(any())).thenReturn(responseMap).thenThrow(EXCEPTION);
+
+        assertThatThrownBy(() -> consumer.accept(allRequests)).isEqualTo(EXCEPTION);
+
+        // assert requests made by client alpha are served
+        assertSanityOfResponse(alphaRequestList, ImmutableMap.of(alpha, ImmutableList.of(responseMap.get(alpha))));
+
+        LockWatchEventCache alphaCache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(alpha);
+        verify(alphaCache).lastKnownVersion();
```

Comment:
nit: do we care that this is exactly 1? Feels like a bit of an implementation detail